content/en/docs/concepts/workloads/controllers/daemonset.md

```diff
@@ -201,6 +201,63 @@ them according to its `updateStrategy`.
 
+While working with the rolling updates of the DaemonSets, you will come across two important parameters: `maxSurge` and `maxUnavailable`.
+
+### maxSurge
+
+If you set this parameter to a number above 0, you allow the DaemonSet controller
+to create temporary duplicate Pods during a rollout.
+Because a DaemonSet aims to have one healthy Pod per node, this field also implies
+the number of _nodes_ that can have a duplicate Pod on them at any one moment
+during the rollout.
+While performing a rolling update, Kubernetes will keep creating new pods to replace the old ones. 
+`maxSurge` helps you to regulate the surge of additional pods beyond the desired count. 
+
+#### maxSurge and Pod failures
+
+During a DaemonSet rollout, one of the old pods maybe become unavailable for some reason
+(for example: the `Ready` condition transitions to false, the  Pod is evicted, or the node is drained).
+If that occurs, the DaemonSet controller immediately creates a replacement Pod on that node,
+without considering surge limits. The replacement Pod uses the current (new) Pod template, not
+the previous one.
+
+### maxUnavailable
+
+This parameter defines the maximum number of pods, belonging to a DaemonSet,
+that can be unavailable while performing an update.
+In simpler terms, they are being deleted and not yet replaced by new pods, during the update process. 
+`maxUnavailable` specifies either the maximum number or maximum proportion of pods that can
+be (voluntarily) [disrupted](/docs/concepts/workloads/pods/disruptions/#voluntary-and-involuntary-disruptions)
+at any particular moment during a rollout.
+
+#### maxUnavailable and Pod failures
+
+During a DaemonSet rollout, one of the old pods maybe become unavailable for some reason
+(for example: the `Ready` condition transitions to false, the  Pod is evicted, or the node is drained).
+If that occurs, the DaemonSet controller immediately creates a replacement Pod on that node,
+and avoids removing other old Pods until the number of unavailable Pods fits inside the
+`maxUnavailable` threshold.
+
+### Example {#updating-a-daemonset-example}
+
+Suppose the value of `maxUnavailable` is set to 1 and the desired count is 5,
+Kubernetes can delete and replace pods one at a time, ensuring the fact that there should always be at least 4 pods available.
+This parameter is useful to maintain the desired level of availability during updates.
+
+You need to set `.spec.updateStrategy.rollingUpdate.maxUnavailable` (default to 1), `.spec.minReadySeconds` (default to 0) and `.spec.updateStrategy.rollingUpdate.maxSurge` (defaults to 0).
+The default value of `maxSurge` is 0 and the default value of `maxUnavailable` is 25%. For 
+`maxUnavailable`,  at most this many nodes that should be running the daemon pod can have
+their pods stopped for an update at any given time.
+
+```yaml
+spec:
+  updateStrategy:
+      type: RollingUpdate
+      rollingUpdate:
+        maxSurge: 1
+        maxUnavailable: 1
+```
```

Comment:
and then maybe we can describe a bit how the deployment will behave with regards to these features?